File: data/multilang_comparison/program_julia/program_julia/src/func43.jl
include("program_julia_head.jl")
function func43(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   loopLimit11 = (50)/4 + 1
   for i in 1:loopLimit11 
   end
   array30 = nothing
   if pCounter > 1
       array30 = vars.data[pCounter]
       pCounter -= 1
       array30.refC += 1
   else
       array30 = JArray(zeros(Int, 286), 286, 1, 30)
   end
   return array30;
end

File: data/multilang_comparison/program_cpp_array/src/func43.cpp
#include "program_cpp_array.hpp" 
Array* func43(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array19;
   if (pCounter > 0) {
      array19 = vars->data[--pCounter];
      array19->refC++;
      DEBUG_COPY(array19->id);
   } else {
      array19 = new Array();
      array19->size = 814;
      array19->refC = 1;
      array19->id = 19;
      array19->data = new unsigned int[array19->size]();
      DEBUG_NEW(array19->id);
   }
   return array19;
}

